hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/ClientConnectionManagerImpl.java

```diff
@@ -318,7 +319,22 @@ public Connection getActiveConnection(Address target) {
         if (target == null) {
             return null;
         }
-        return activeConnections.get(target);
+
+        Address ipAddress = getIpAddress(target);
+
+        if (ipAddress == null) {
+            return null;
+        }
+
+        return activeConnections.get(ipAddress);
+    }
+
+    private Address getIpAddress(Address target) {
+        try {
+            return new Address(target.getInetAddress().getHostName(), target.getPort());
+        } catch (UnknownHostException e) {
+            return null;
```

Comment:
I don't currently see any unwanted side effects of returning `null` here but maybe it leaves a window open to misinterpretation in the future? Maybe wrapping in a `HazelcastException` and rethrowing is a more clean approach?